oidc-api/src/test/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandlerTest.java

```diff
@@ -1693,6 +1694,136 @@ void shouldErrorIfIdTokenHasIncorrectClient() throws JOSEException {
                             BASE_AUDIT_USER.withSessionId(SESSION_ID),
                             pair("description", expectedErrorObject.getDescription()));
         }
+
+        @Nested
+        class BrowserSessionIdJourney {
+
+            APIGatewayProxyResponseEvent setupWithExistingSession(
+                    Session existingSession, String browserSessionIdFromCookie) {
+                when(configService.isBrowserSessionCookieEnabled()).thenReturn(true);
+                when(sessionService.getSessionFromSessionCookie(any()))
+                        .thenReturn(Optional.ofNullable(existingSession));
+                when(sessionService.createSession())
+                        .thenReturn(
+                                new Session("a-new-session-id")
+                                        .withBrowserSessionId("a-new-browser-session-id"));
+
+                Map<String, String> requestParams = buildRequestParams(null);
+                APIGatewayProxyRequestEvent event = withRequestEvent(requestParams);
+                event.setRequestContext(
+                        new ProxyRequestContext()
+                                .withIdentity(
+                                        new RequestIdentity().withSourceIp("123.123.123.123")));
+                if (browserSessionIdFromCookie != null) {
+                    event.setHeaders(Map.of("browser-session-id", browserSessionIdFromCookie));
+                }
+
+                return makeHandlerRequest(event);
+            }
+
+            String browserSessionIdCookieFromResponse(APIGatewayProxyResponseEvent response) {
+                return response.getMultiValueHeaders().get(ResponseHeaders.SET_COOKIE).get(2);
+            }
+
+            @Test
+            void noExistingSessionAndNoBrowserSessionIdInHeaders() {
```

Comment:
ideally test names describe a behaviour we're testing for, rather than a scenario